servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -33,6 +33,10 @@ class Http4sServlet(service: HttpService,
   // micro-optimization: unwrap the service and call its .run directly
   private[this] val serviceFn = service.run
 
+  object ServletRequestKeys {
+    val HttpSession: AttributeKey[HttpSession] = AttributeKey.http4s("request.httpSession")
```

Comment:
I think this should be an `Option[HttpSession]`, so we don't create sessions where none exist.  Then the key could be populated with `Option(req.getSession(false))`.